odoo/addons/base/models/res_lang.py

```diff
@@ -94,6 +99,34 @@ def _check_grouping(self):
             except Exception:
                 raise ValidationError(warning)
 
+    @api.depends('date_format')
+    def _compute_date_format_preview(self):
```

Comment:
I would mention in the docstring that we use direct babel calls instead of tooling methods because the language may not be active yet.
(Same for the time formatting).

Reply:
Hello @awa-odoo 

Changes are done.

Thanks